File: frontend/src/types/room.ts
export type EntityStatus = 'ACTIVE' | 'DEACTIVATED'

export interface Building {
  id: string
  name: string
  status: EntityStatus
}

export interface Floor {
  id: string
  buildingId: string
  name: string
  status: EntityStatus
}

export interface EquipmentType {
  id: string
  name: string
  status: EntityStatus
}

export interface SeatingArrangement {
  id?: string
  name: string
  maxCapacity: number
}

export interface RoomSummary {
  id: string
  name: string
  building: Building
  floor: Floor
  status: EntityStatus
}

export interface Room extends RoomSummary {
  version: number
  seatingArrangements: SeatingArrangement[]
  equipmentTypeIds: string[]
}

export interface RoomCreateRequest {
  name: string
  floorId: string
  seatingArrangements: SeatingArrangement[]
  equipmentTypeIds?: string[]
}

export interface RoomUpdateRequest extends RoomCreateRequest {
  version: number
}

export interface ProblemFieldError {
  field: string
  message: string
}

export interface Problem {
  title: string
  status: number
  detail: string
  errors?: ProblemFieldError[]
}

export type StatusFilter = 'active' | 'deactivated' | 'all'

File: backend/src/main/java/at/mci/igp/raumlotse/domain/Room.java
package at.mci.igp.raumlotse.domain;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.UuidGenerator;

@Entity
@Table(name = "room")
public class Room {

    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;

    private String name;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "floor_id", nullable = false)
    private Floor floor;

    @Enumerated(EnumType.STRING)
    private EntityStatus status = EntityStatus.ACTIVE;

    @Version
    private Long version;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;

    @OneToMany(mappedBy = "room", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<SeatingArrangement> seatingArrangements = new ArrayList<>();

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "room_equipment",
            joinColumns = @JoinColumn(name = "room_id"),
            inverseJoinColumns = @JoinColumn(name = "equipment_type_id"))
    private List<EquipmentType> equipmentTypes = new ArrayList<>();

    protected Room() {
    }

    public Room(String name, Floor floor) {
        this.name = name;
        this.floor = floor;
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Floor getFloor() {
        return floor;
    }

    public void setFloor(Floor floor) {
        this.floor = floor;
    }

    public EntityStatus getStatus() {
        return status;
    }

    public void setStatus(EntityStatus status) {
        this.status = status;
    }

    public Long getVersion() {
        return version;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }

    public List<SeatingArrangement> getSeatingArrangements() {
        return seatingArrangements;
    }

    public List<EquipmentType> getEquipmentTypes() {
        return equipmentTypes;
    }

    public void setEquipmentTypes(List<EquipmentType> equipmentTypes) {
        this.equipmentTypes = equipmentTypes;
    }

    public void replaceSeatingArrangements(List<SeatingArrangement> replacement) {
        seatingArrangements.clear();
        for (SeatingArrangement arrangement : replacement) {
            arrangement.setRoom(this);
            seatingArrangements.add(arrangement);
        }
    }
}
